src/main/java/uk/ac/manchester/beehive/tornado/plugins/ui/settings/TornadoSettingState.java

```diff
@@ -40,6 +40,7 @@ public class TornadoSettingState implements PersistentStateComponent<TornadoSett
     @OptionTag(converter = JdkConverter.class)
     public Sdk JdkPath;
     public int parameterSize;
+    public String tensorShapeDimensions;
```

Comment:
Should we represent the dimension directly in an `int` value, instead of a `String`?

Reply:
yes, makes sense. I decided to keep it as String because I was getting exceptions due to serialization for the `TornadoSettingState.java` class. I will keep a note to refactor it in future PRs.